### DISCUSSION TITLE
What OS are you on and why?

### DISCUSSION THREAD

**Discussion by fumitoh** [actuarialopensource/website#40] [Team Posts]
I've been on Windows for long time because of Excel and its dominance in finance, 
but the majority of open source developers and data scientist seems to be on Linux (or on Darwin?)

**Comment by keluc** [comment]
Same arguments here

**Comment by lewisfogden** [comment]
I'm using Windows but have Ubuntu running on a virtual machine on it (using WSL), would have Linux running on my laptop but had driver issues which was frustrating.  I find a lot of development work is better/easier on Linux, particularly using the terminal, however Windows isn't too bad for some tooling, and it has plenty of IDE options.

VSCode integrates with WSL so you can write in Windows and run in Linux - sample below:

(WSL = Windows Subsytem for Linux)

![wsl_vscode](https://user-images.githubusercontent.com/8605587/117059917-3da55d80-ad18-11eb-8697-d203a6d381c1.png)

**Comment by fumitoh** [comment]
That's good to know. I tested WSL some time ago but didn't keep using it. I use PyCharm and it seems to support Python on WSL as well. WSL2 is going to make GUI programs on guest Linux appear like normal Windows windows. 
https://devblogs.microsoft.com/commandline/the-initial-preview-of-gui-app-support-is-now-available-for-the-windows-subsystem-for-linux-2/

**Comment by pdavidsonFIA** [comment]
Pycharm also plays with WSL.
You can also set up WSL to run programs to an X-sessuon, then use an X host in windows (I use VcXsrv). Then u can run any Linux gui app in a window.
Once the new WSL patch comes out this will be redundant:) Note WSL2 already out - gui patch in development

**Comment by alecloudenback** [comment]
I have been on Mac almost exclusively since late 2019 and very much enjoy it. I previously primarily used Windows for the prior ~7 years (Mac before that). Throughout I've dabbled with different linux distros.

**Comment by jimbrig** [comment]
Between WSL (and Docker WSL Integrations), WinGet Package Manager, VSCode and Visual Studio (and Azure Data Studio), Windows Terminal, PowerToys, Office Suite, and the simplicity of PowerShell Windows has really made a surge for development focused OS configurations and setups in my opinion.

**Comment by MatthewCaseres** [comment]
I got an M1 mac when they came out and switched from Windows. 

Remote development is becoming more of a thing lately with GitHub codespaces. Last place I worked at had a lot of people using AWS Sagemaker Studio notebooks. Those development environments are easy to get started but there was no syntax highlighting or jumping to definitions.

Instead of Sagemaker's notebooks I would SSH into an Ubuntu EC2 instance with GPU and develop from there in VS Code. Most data scientists didn't like this amount of operational overhead and stuck with the managed notebooks from AWS.
